scalableminds/webknossos
# add protected and private modifiers to DAO hierarchy

Re-generating Tables.scala from a changed SQL schema led to a ton of unneeded recompilation.

This PR adds protected/private access modifiers in the DAO hierarchy to make everything that has stuff from Tables.scala in the signature protected/private in the DAO scope. This was implicitly already the case for most methods. The exception was parse, where two non-private usages existed in associated DAOs (SharedAnnotationsDAO, UserTeamRolesDAO). I merged these into the main DAOs AnnotationsDAO and UserDAO respectively to avoid having to make parse public.

### URL of deployed dev instance (used for testing):
- https://daoprivate.webknossos.xyz

### Steps to test:
- CI should be enough to ensure we didn’t break anything
- To see the speedup, add a new column to a table in schema.sql, execute yarn refresh-schema and recompile → Should not recompile over a hundred source files

------
